### ISSUE TITLE
create-sibling should fail sooner than later when remote host has no annex

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#1192]
We do require git-annex on remote box if publishing git annex repositories, so we could 'git annex init' them there to get uuid.  If we are creating with -r, it just keeps failing for all of them instead of ideally first sensing that annex is not available and quitting without performing any action (so we could rerun it after having annex installed and have no side-effects of partially configured repositories which would not be reconfigured)

Sample error:
```
[INFO   ] Creating target dataset bids/dbic/dbic-animals/derivatives/mriqc at /inbox/BIDS/dbic/dbic-animals/derivatives/mriqc 
[ERROR  ] Initialization of remote git annex repository failed at /inbox/BIDS/dbic/dbic-animals/derivatives/mriqc.
| Error: CommandError: command '['ssh', '-o', u'ControlPath=/home/mvdoc/.cache/datalad/sockets/rolando.cns.dartmouth.edu', u'rolando.cns.dartmouth.edu', "'git'", "'-C'", u"'/inbox/BIDS/dbic/dbic-animals/derivatives/mriqc'", "'annex'", "'init'"]' failed with exitcode 1
| Failed to run ['ssh', '-o', u'ControlPath=/home/mvdoc/.cache/datalad/sockets/rolando.cns.dartmouth.edu', u'rolando.cns.dartmouth.edu', "'git'", "'-C'", u"'/inbox/BIDS/dbic/dbic-animals/derivatives/mriqc'", "'annex'", "'init'"] under None. Exit code=1. out= err=git: 'annex' is not a git command. See 'git --help'.
|  [cmd.py:run:370]
``

**Comment by mih** [comment]
And do this as part of a general remote sensing step that also gets a git version....

**Comment by mih** [comment]
Once #1241 is merged, this will be trivial to implement.

**Comment by mih** [comment]
Fix in #1237
